Create and Remove Books

CollinsTatang opened the pull request:
**This branch required the following:**
- Let's start by configuring the Redux Store (/src/redux/configureStore.js) and importing the necessary methods from Redux.
- Write your book's actions and reducer (in /src/redux/books/books.js) using the ducks pattern.
- In your React component responsible for adding new books set the data inputs in the local React state set title and author.
- In your React component responsible for removing books - implement that even by dispatching a corresponding action.

bellom reviewed (changes requested):
## Hi @CollinsTatang  👏 , 

### Great job starting this milestone 👍 

#### You have done wonderfully well setting up linter and good commit messages 🔥, However, you stopped halfway, you need to continue with the task given on the [project requirement page](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/project_create_and_remove_book.md)

# Status: Required Changes! ♻️ 

Seems like you stopped at `configuring the Redux Store`

So kindly attend to the following under the[ Project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/project_create_and_remove_book.md#project-requirements) ;
- `Write your book's actions and reducer (in /src/redux/books/books.js)`
- `In your React component responsible for adding new books set the data inputs in the local React state`
- `In your React component responsible for removing books`

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

CollinsTatang commented:
Sorry, I added the commit for the book's action and reducer but forget to push. 
- I have added the book's action and reducer.
- React component responsible for adding new books sets the data inputs in the local React state.
- React component responsible for removing books.

Emmyn5600 commented on src/components/books/Books.js:
```diff
@@ -1,7 +1,61 @@
     <div>
       <h3>All Books</h3>
+      <div>
+        <ul>
+          { books.map((book) => (
+            <li key={book.id}>
+              <span className="">{book.title}</span>
+              <button id={book.id} type="button" onClick={deleteBookFromStore}>Delete</button>
+            </li>
+          ))}
+        </ul>
+      </div>
+      <form onSubmit={submitToBookStore}>
+        <h4>Add New Book</h4>
+        <input placeholder="Title" value={title} required onChange={setBookTitle} />
+        <br />
+        <br />
+        <input placeholder="Author" value={author} required onChange={setBookAuthor} />
+        <br />
+        <br />
+        <button type="submit">Add Book</button>
+      </form>
     </div>
```
- [ ] Kindly, let us display the author of the book.
![react9](https://user-images.githubusercontent.com/55635977/130784843-925226f3-f1e5-450e-8d3c-a3185ac8d72c.png)

Emmyn5600 reviewed (changes requested):
## STATUS: CHANGES REQUESTED :recycle:

Hi, @CollinsTatang 

Great work on implementing the changes requested by the previous reviewer, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
   - The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Categories still under construction:heavy_check_mark:
- Booklist renders the list of books:heavy_check_mark:
- Add book works perfectly.:heavy_check_mark:
- Remove book works perfectly.:heavy_check_mark:

Changes required :writing_hand:  

      * #### Please check the comments under the review.
       
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

mikenath223 reviewed (approved):
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
